jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/ee/rs/core/headers/HttpHeadersTest.java

```diff
@@ -67,6 +70,39 @@ public String headersGet() {
     return sb.toString();
   }
 
+  @GET
+  @Path("/contains-headers")
+  public String containsHeadersGet() {
+    sb = new StringBuffer();
+    List<String> myHeaders = Arrays.asList("Accept", "Content-Type", "Header3");
+
+    try {
+        assertTrue(hs.containsHeaderString("accept", "text/html"::equalsIgnoreCase));
+        sb.append("Header: accept contains text/html; ");
+      
+        //Verify Predicate and separator character usage
+        assertTrue(hs.containsHeaderString("Accept", ",", "Text/html;Level=1"::equalsIgnoreCase));
+        sb.append("Header: accept contains text/html;level=1; ");
+        
+        //Verify incorrect separator character fails
+        assertFalse(hs.containsHeaderString("Accept", ";", "text/html;level=1"::equals));
+        sb.append("Incorrect separator character fails as expected; ");
+        
+        //Verify white space in parameter not trimmed
+        assertFalse(hs.containsHeaderString("header3", "**", "value3"::equalsIgnoreCase));
```

Comment:
I get the following:
```
java.util.regex.PatternSyntaxException: Dangling meta character '*' near index 0
**
^
	at java.base/java.util.regex.Pattern.error(Pattern.java:2095)
	at java.base/java.util.regex.Pattern.sequence(Pattern.java:2270)
	at java.base/java.util.regex.Pattern.expr(Pattern.java:2136)
	at java.base/java.util.regex.Pattern.compile(Pattern.java:1836)
	at java.base/java.util.regex.Pattern.<init>(Pattern.java:1467)
	at java.base/java.util.regex.Pattern.compile(Pattern.java:1092)
	at java.base/java.lang.String.split(String.java:3208)
	at java.base/java.lang.String.split(String.java:3290)
```
I'd better use that exception than consume it and return false.

Reply:
Yeah, that looks like a glob style pattern. A match all would be `.*`.

Reply:
Agreed.  Changed to "::" separator